bytes/last_index_of_needle.ts

```diff
@@ -20,7 +20,7 @@
 export function lastIndexOfNeedle(
   source: Uint8Array,
   needle: Uint8Array,
-  start = source.length - 1,
+  start: number = source.length - 1,
```

Comment:
Is this required? `deno doc --lint bytes/last_index_of_needle.ts` reports no errors before this change.

Reply:
I think there's a bug in `deno doc --lint` related to this. I'm running a separate analyzer (it is an optimization to type checking).

See here how the type is missing: https://deno.land/std@0.210.0/bytes/last_index_of_needle.ts?s=lastIndexOfNeedle

Reply:
Oh, I see. Perhaps we should fix that bug instead of accounting for it. WDYT?

Reply:
I opened https://github.com/denoland/deno_doc/issues/435

Doing a big refactor of everything at the moment (which is what helped spot this), so I'll add a test and investigate in there later.

Reply:
Perhaps let's continue to include these explicit types on arguments with default values. It'll fix the docs sooner. We can then come behind later on and clean these up. I can create an issue once this PR is merged.

Reply:
I think this is all the public API that's affected. Everything else seems to be good. After this PR I have Deno type checking a subset of the code properly in the new registry.